toolchain/driver/clang_subcommand.cpp

```diff
@@ -51,8 +51,10 @@ auto ClangSubcommand::Run(DriverEnv& driver_env) -> DriverResult {
   // Don't run Clang when fuzzing, it is known to not be reliable under fuzzing
   // due to many unfixed issues.
   if (driver_env.fuzzing) {
-    *driver_env.error_stream
-        << "error: cannot run `clang` subcommand productively when fuzzing\n";
+    CARBON_DIAGNOSTIC(
+        ClangFuzzingDisallowed, Error,
```

Comment:
Maybe this is a TODO that only makes sense to address in the future, but I feel like we should have diagnostics declaratively state the level of source location they should be associated with, rather than that being a property of which emitter they are passed to.

Mostly want to mention that in case it motivates a different API design here where maybe we have a single emitter but different emit functions that use different diagnostic types to select the location accepted? Dunno, that might not end up working well. I'm not really bothered by having different emitters, I'd just like there to be some declarative connection eventually so that we can't accidentally use the same diagnostic both with- and without location info. (Or try to wordsmith the same message to work in both cases, etc.)

Reply:
If I understood discussion correctly last night, you're saying the `CARBON_DIAGNOSTIC` macro should include a parameter which indicates the location type that it's meant to be used with. From my perspective, that's an expensive addition, and typically we have the `CARBON_DIAGNOSTIC` adjacent to the use, so would also offer marginal utility.

If you restructure how diagnostics are declared, perhaps we could evaluate whether it'd be clearer to indicate that in the given context, instead of preemptively? Even in such a future, we may find that most files only emit with one location type.

Reply:
Yeah... And this is just a forward looking thing. I've suggested some candidate TODO text above on the `NoLoc` diagnostic emitter that maybe captures this?